### DISCUSSION TITLE
Are firewall rules supported on the storage account for durable functions?

### DISCUSSION THREAD

**Discussion by JeroenJSmit** [Azure/azure-functions-durable-extension#2116] [Q&A]
We have our function apps (V4, .Net 6) deployed in VNets (subnets) and would like to apply general IP restrictions to the storage account connected to the function app (the one for for durable function state, blob leases, etcetera). Is that supported? I cannot read anything on that explicitely.

Some documentation mentions private endpoints, some information is on service endpoints.
We would just like to enable the following
![image](https://user-images.githubusercontent.com/4172510/158807791-0c5c7e01-6b6d-4174-9927-e8989f8c2630.png)

**Comment by olitomlinson** [comment]
@JeroenJSmit 

I haven't tested this, but I don't see why adding a Firewall IP range would impact the functioning of the TaskHub in your storage Account. All resources in the VNET should be able to communicate with the Storage Account (if you connect the Storage Account into the VNET as per the above configuration)

Ultimately, I would recommend giving it a quick go to build certainty - Add your IP range, (or even a fictitious IP range if you don't have one yet) and try running a sample Hello World Durable Functions App connected to this Storage Account. 

Chances are if the Function App can't communicate successfully to the TaskHub in the Storage Account, your App will struggle to start-up successfully, so I would expect a whole bunch of errors in App Insights or the Portal etc.

**Reply by JeroenJSmit** [reply]
Thanks for the response!

We raised a support ticket in 2020 where we ran into some issues and eventually got the answer that there are no guarantees nor SLA and we were wondering if that is still the case. 

Reading https://docs.microsoft.com/en-us/azure/azure-functions/configure-networking-how-to#restrict-your-storage-account-to-a-virtual-network it seems to suggest additional configuration is necessary to restrict a storage account to a virtual network. Is that also the case when we only want to use service endpoints for whitelisting subnet instead of private endpoints?

**Reply by olitomlinson** [reply]
@JeroenJSmit 

I was able to connect Function App (premium plan) to a VNET (subnet1)

![image](https://user-images.githubusercontent.com/4224880/160501688-7c226567-aca9-485b-8bf0-96afc2c8c742.png)

and then connect a Storage Account via Private Endpoint to the same VNET (subnet 2).

![image](https://user-images.githubusercontent.com/4224880/160501567-c52fadf9-bdf5-469f-9cee-19945707e6f8.png)

and then configured the Storage Account to only allow selected networks (not public) but didn't chose any particular Network. I also added my own IP address so I could inspect the Storage Account using storage explorer.

![image](https://user-images.githubusercontent.com/4224880/160501979-784d5563-5af4-45dc-a4b8-fdcf9ad92949.png)

Can confirm the TaskHub residing in the locked-down Storage Account is working just fine from the Function App.

![image](https://user-images.githubusercontent.com/4224880/160502099-c2ddeca6-7d91-4723-aa97-6201418adc01.png)

**Comment by JeroenJSmit** [comment]
We know it is working, it has been for a while but we want to know what is officially supported.

**Reply by olitomlinson** [reply]
I don't see why it wouldn't be officially supported. Its just an infrastructure thing, between the Function App and the Storage Account (It being a TaskHub should be of no concern)

@cgillum any thoughts?
